libnavigation-core/src/main/java/com/mapbox/navigation/core/history/MapboxHistoryRecorder.kt

```diff
@@ -34,52 +34,65 @@ class MapboxHistoryRecorder internal constructor(
      * @return absolute path to the directory with history files.
      *     returns null when there is no file directory
      */
-    fun fileDirectory(): String? {
-        return if (historyRecorderOptions.enabled) {
-            historyFiles.absolutePath(historyRecorderOptions)
-        } else {
-            null
-        }
+    fun fileDirectory(): String? = historyFiles.absolutePath(historyRecorderOptions)
+
+    /**
+     * Starts history recording session.
+     * If history recording is already started - does nothing.
+     * To save history and get a file path call [MapboxHistoryRecorder.stopRecording].
+     */
+    fun startRecording() {
+        checkRecorderInitialized()
+        historyRecorderHandle?.apply { startRecording() }
     }
 
     /**
-     * When called, history recording will be paused and all records will be added to the file.
+     * When called, history recording will be stopped and all records will be added to the file.
      * The path to the file will be sent to the [SaveHistoryCallback.onSaved] interface.
-     * The next call to to this function will include a new file with new events.
+     * If history recording was not started before - [SaveHistoryCallback.onSaved] will be called
+     * with `null` filepath.
+     * A new recording session is not started automatically after this call, if it is required -
+     * need to call [MapboxHistoryRecorder.startRecording].
```

Comment:
nit: i think we can remove the comment of what it doesn't do, and say what it does.

Also, can we verify that history startRecording can be called before the SaveHistoryCallback finishes? Or is this something the developer would need to implement?

```suggestion
     * If history recording was not started, [SaveHistoryCallback] is called with a `null` file path.
     * [MapboxHistoryRecorder.startRecording] can be called before the [SaveHistoryCallback] completes.
```

Reply:
> can we verify that history startRecording can be called before the SaveHistoryCallback finishes

can we? 
cc @mskurydin

Reply:
@korshaknn 
cc @mskurydin 
Yes, you can verify that `startRecording()` can be called before the `SaveHistoryCallback` finishes.
Actually, you shouldn't wait for the `SaveHistoryCallback` finish. You can call `startRecording()` right after `stopRecording(...)` - this is correct if you want to achieve the same behavior as with "autorecording" mode - going this way you will not miss events. And if you call `startRecording()` only in callback - the callback execution can be delayed, and you could miss some events.